app/src/test/java/co/nimblehq/compose/crypto/ui/screens/detail/DetailScreenTest.kt

```diff
@@ -0,0 +1,178 @@
+package co.nimblehq.compose.crypto.ui.screens.detail
+
+import androidx.activity.compose.setContent
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.*
+import androidx.navigation.*
+import co.nimblehq.compose.crypto.R
+import co.nimblehq.compose.crypto.domain.usecase.GetCoinDetailUseCase
+import co.nimblehq.compose.crypto.domain.usecase.GetCoinPricesUseCase
+import co.nimblehq.compose.crypto.extension.toFormattedString
+import co.nimblehq.compose.crypto.test.MockUtil
+import co.nimblehq.compose.crypto.ui.navigation.AppDestination
+import co.nimblehq.compose.crypto.ui.screens.BaseViewModelTest
+import co.nimblehq.compose.crypto.ui.screens.MainActivity
+import co.nimblehq.compose.crypto.ui.uimodel.toUiModel
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.shadows.ShadowToast
+import kotlin.math.abs
+
+@RunWith(RobolectricTestRunner::class)
+@ExperimentalCoroutinesApi
+class DetailScreenTest : BaseViewModelTest() {
+
+    @get:Rule
+    val composeAndroidTestRule = createAndroidComposeRule<MainActivity>()
+
+    private val marketCapTitle: String
+        get() = composeAndroidTestRule.activity.getString(R.string.detail_market_cap_title)
+
+    private val allTimeHighTitle: String
+        get() = composeAndroidTestRule.activity.getString(R.string.detail_all_time_high_title)
+
+    private val allTimeLowTitle: String
+        get() = composeAndroidTestRule.activity.getString(R.string.detail_all_time_low_title)
+
+    private val errorGeneric: String
+        get() = composeAndroidTestRule.activity.getString(R.string.error_generic)
+
+    private val mockGetCoinDetailUseCase = mockk<GetCoinDetailUseCase>()
+    private val mockGetCoinPricesUseCase = mockk<GetCoinPricesUseCase>()
+
+    private lateinit var viewModel: DetailViewModel
+
+    private var appDestination: AppDestination? = null
+
+    private val coinDetailUiModel = MockUtil.coinDetail.toUiModel()
+
+    @Before
+    fun setUp() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flowOf(MockUtil.coinDetail)
+
+        composeAndroidTestRule.activity.setContent {
```

Comment:
We can `initViewModel()` in `setup`
```suggestion
        every { mockGetCoinDetailUseCase.execute(any()) } returns flowOf(MockUtil.coinDetail)
        initViewModel()
        composeAndroidTestRule.activity.setContent {
```